### PR TITLE
refactor: optimization of activation processing

### PR DESCRIPTION
**Pull request by alex-gilin**

### CODE REVIEW

**Review by marufrasully — APPROVED** [review]
Thanks @alex-gilin 
Looks good to me.

File: packages/vscode-ui5-language-assistant/src/extension.ts [hunk 1]
```diff
@@ -55,24 +55,15 @@ let client: LanguageClient;
...
-  // register semantic token provider
-  context.subscriptions.push(
-    languages.registerDocumentSemanticTokensProvider(
-      { language: "xml" },
-      bindingSemanticTokensProvider,
-      bindingLegend
-    )
-  );
-
+async function init(context: ExtensionContext): Promise<void> {
```
Comment (voicis) [inline packages/vscode-ui5-language-assistant/src/extension.ts:58]: Why is this function async? There are no awaits and it does not return a promise.

Comment (alex-gilin) [reply packages/vscode-ui5-language-assistant/src/extension.ts]: done

**Comment by marufrasully** [comment]
This PR handled here https://github.com/SAP/ui5-language-assistant/pull/724.

**Pull request closed without merging**